code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -1562,46 +1562,105 @@ All effects don't start immediately, but rather get worse over time; the rate is
 	glass_icon_state = "sidecar"
 	glass_name = "Sidecar"
-	glass_desc = "The one ride you’ll gladly give up the wheel for."
+	glass_desc = "The one ride you'll gladly give up the wheel for."
 
 /datum/reagent/consumable/ethanol/between_the_sheets
 	name = "Between the Sheets"
 	id = "between_the_sheets"
-	description = "A provocatively named classic."
+	description = "A provocatively named classic. Funny enough, doctors recommend drinking it before taking a nap."
 	color = "#F4C35A"
 	boozepwr = 80
 	taste_description = "seduction"
 	glass_icon_state = "between_the_sheets"
 	glass_name = "Between the Sheets"
-	glass_desc = "A provocatively named classic."
+	glass_desc = "The only drink that comes with a label reminding you of Nanotrasen's zero-tolerance promiscuity policy."
+
+/datum/reagent/consumable/ethanol/between_the_sheets/on_mob_life(mob/living/L)
+	..()
+	if(L.IsSleeping())
+		if(L.bruteloss && L.fireloss) //If you are damaged by both types, slightly increased healing but it only heals one. The more the merrier wink wink.
+			if(prob(50))
+				L.adjustBruteLoss(-0.25)
+			else
+				L.adjustFireLoss(-0.25)
+		else if(L.bruteloss && !L.fireloss) //If you have only one, it still heals but not as well.
+			L.adjustBruteLoss(-0.2)
+		else if(!L.bruteloss && L.fireloss)
+			L.adjustFireLoss(-0.2)
 
 /datum/reagent/consumable/ethanol/kamikaze
 	name = "Kamikaze"
 	id = "kamikaze"
-	description = "Divinely windy."
+	description = "Science has yet to understand the seemingly mystical properties of this drink. Not only does it damage the individual, but it also damages those nearby the individual can see."
 	color = "#EEF191"
 	boozepwr = 60
 	taste_description = "divine windiness"
 	glass_icon_state = "kamikaze"
 	glass_name = "Kamikaze"
-	glass_desc = "Divinely windy."
+	glass_desc = "A drink you don't exactly make friends with."
+	var/mutable_appearance/explosion_overlay
+
+/datum/reagent/consumable/ethanol/kamikaze/on_mob_add(mob/living/L)
+	explosion_overlay = mutable_appearance('icons/effects/effects.dmi', "explosion_indef")
+	explosion_overlay.alpha = 50
+	L.add_overlay(explosion_overlay)
+
+/datum/reagent/consumable/ethanol/kamikaze/on_mob_life(mob/living/L)
+	..()
+	var/victimnum = 0
+	explosion_overlay.alpha = CLAMP(explosion_overlay.alpha,0,100)
+	for(var/mob/living/victim in oview(1,L)) //every victim takes flat 1 damage, but the user takes 2 per individual affected.
+		victim.adjustFireLoss(1)
+		L.adjustFireLoss(2)
+		explosion_overlay.alpha += 1
+		victimnum++
+	if(!victimnum) //No one is nearby so you take decreased damage.
+		L.adjustFireLoss(1)
+		explosion_overlay.alpha -= 1
+		if(explosion_overlay.alpha <= 0) //can be -1 until next tick
+			metabolization_rate = (initial(metabolization_rate)*5) //The overlay is gone so the drink rapidly dissapates. Congrats, you defused the -explosive- i mean drink!
+
+/datum/reagent/consumable/ethanol/kamikaze/on_mob_delete(mob/living/L)
+	L.cut_overlay(explosion_overlay)
```

Comment:
The overlay is being removed or updating its alpha properly.

Reply:
I tested this, the drink metabolizing does not remove the overlay, and I can't distinguish its alpha visibly changing from the default value.

Not an expert in how overlays work but you may need need to cut, edit alpha, and add again? Pester #coderbus, maybe someone knows.

Reply:
ehhh I'll just play with it later, this pr has gone far too long.